compiler/nimconf.nim

```diff
@@ -238,6 +238,26 @@ proc getSystemConfigPath*(conf: ConfigRef; filename: RelativeFile): AbsoluteFile
     if not fileExists(result): result = p / RelativeDir"etc/nim" / filename
     if not fileExists(result): result = AbsoluteDir"/etc/nim" / filename
 
+const nimRepoDirCT = currentSourcePath / ".." / ".." # this should be exposed somewhere
+
+proc getNimRoot*(conf: ConfigRef): AbsoluteDir =
+  ## returns root nim dir; works both with installed nim and local nim git repo.
+  ## This can be easily overridden if needed with NIM_ROOT_DIR environment variable
+  ## The result should satisfy: $result/doc/basicopt.txt exists
+  if conf.nimRootDir.isEmpty:
+    result = getEnv("NIM_ROOT_DIR").AbsoluteDir
```

Comment:
We discussed this and I won't accept it, sorry. One more reason why I dislike environment vars: Some fellow developer can set it locally, forget about this setting, and then is confused about why the test results on the CIs are different.

Reply:
removed; the logic still works with both installed nim and local repo nim without `NIM_ROOT_DIR` as I was explaining in https://github.com/nim-lang/Nim/pull/13117#issuecomment-573477159